### PR TITLE
gh-127949: deprecate `asyncio.set_event_loop_policy`

### PR DESCRIPTION
**Pull request by kumaraditya303**
I am skipping news for now, I'll add docs and what's new when all the work is completed at the end to avoid having many news entry for the same. 

* Issue: gh-127949

----
📚 Documentation preview 📚: https://cpython-previews--128024.org.readthedocs.build/

### CODE REVIEW

**Review by asvetlov — APPROVED** [review]
LGTM

File: Doc/library/asyncio-policy.rst [hunk 1]
```diff
@@ -46,6 +46,10 @@ for the current process:
 
    If *policy* is set to ``None``, the default policy is restored.
 
+   .. deprecated:: 3.14
```
Comment (picnixz) [inline Doc/library/asyncio-policy.rst]:
```suggestion
   .. deprecated:: next
```

File: Lib/asyncio/events.py [hunk 2]
```diff
@@ -21,6 +23,7 @@
 import subprocess
 import sys
 import threading
+import warnings
```
Comment (picnixz) [inline Lib/asyncio/events.py:26]: We generally use local imports for deprecation warnings. WDYT?

**Review by picnixz — COMMENTED** [review]
Small nits.

File: Lib/asyncio/events.py (hunk 2 above)
Comment (kumaraditya303) [reply Lib/asyncio/events.py:26]: The upcoming PRs will add more warnings so I kept it at module level

**Review by picnixz — APPROVED** [review]

**Pull request merged**